modules/home-environment.nix

```diff
@@ -105,6 +105,10 @@ let
     };
   };
 
+  getEnvOrVar = var: let val = builtins.getEnv var;
```

Comment:
I'm not sure I understand this 😕

Wouldn't this set, for example, the `homeDirectory` option to literally `"$HOME"` if `getEnv "HOME"` returns an empty string? If so then it would break any code that uses this option in a place that doesn't do shell like environment variable expansion.

Reply:
Right - you could only use this in something that is eventually run by the activation script.

There's a probably a better way to handle this - like making all of these paths relative to `$HOME` and forcing expressions to handle the case where homeDirectory is unset manually, but this at least gets restrict-eval working.